api/lib/infrastructure/temporary-storages/InMemoryTemporaryStorage.js

```diff
@@ -0,0 +1,28 @@
+const { v4: uuidv4 } = require('uuid');
+const TemporaryStorage = require('./TemporaryStorage');
+
+class InMemoryTemporaryStorage extends TemporaryStorage {
+
+  constructor() {
+    super();
+    this._client = new Map();
+  }
+
+  async save({ value, expirationDelaySeconds }) {
+    const key = uuidv4();
+    this._client.set(key, value);
+
+    const thisClient = this._client;
+    setTimeout(function() {
+      thisClient.delete(key);
+    }, expirationDelaySeconds * 1000);
```

Comment:
**remarque**
Ca ne change rien alors je propose de NE PAS faire le changement, mais pour info : 
je crois qu'on peut écrire ça aussi avec une arrow function pour éviter de créer la constante `thisClient`.
Quelqu'un confirme ? 
```
setTimeout(() => {
      this._client.delete(key);
    }, expirationDelaySeconds * 1000);
```

Reply:
Tout à fait, ça fonctionnerati à peu près pareil (avec une subtilité, si `this._client` change de valeur dans l'intervalle). Et effectivement dans Pix on utilise normalement plutôt les arrow functions qu'une sauvegarde de `this` manuelle…

Reply:
Done